featuretools/tests/feature_discovery/test_convertors.py

```diff
@@ -0,0 +1,243 @@
+        ("idx", "Integer", {"index"}),
+        ("f_1", "Double"),
+        ("f_2", "Double"),
+        ("f_3", "NaturalLanguage"),
+    ]
+
+    df = generate_fake_dataframe(
+        col_defs=col_defs,
+    )
+
+    es = EntitySet(id="es")
+    es.add_dataframe(df, df.ww.name, index="idx")
+
+    fdefs = dfs(
+        entityset=es,
+        target_dataframe_name=df.ww.name,
+        trans_primitives=[AddNumeric, TestMultiOutputPrimitive],
+        features_only=True,
+    )
+    assert isinstance(fdefs, list)
+    assert isinstance(fdefs[0], FeatureBase)
+
+    converted_features = set(convert_featurebase_list_to_feature_list(fdefs))
+
+    f1 = LiteFeature("f_1", Double)
+    f2 = LiteFeature("f_2", Double)
+    f3 = LiteFeature("f_3", NaturalLanguage)
+    fadd = LiteFeature(
+        name="f_1 + f_2",
+        tags={"numeric"},
+        primitive=AddNumeric(),
+        base_features=[f1, f2],
+    )
+    fmo0 = LiteFeature(
+        name="TEST_MO(f_3)[0]",
+        tags={"numeric"},
+        primitive=TestMultiOutputPrimitive(),
+        base_features=[f3],
+        idx=0,
+    )
+    fmo1 = LiteFeature(
+        name="TEST_MO(f_3)[1]",
+        tags={"numeric"},
+        primitive=TestMultiOutputPrimitive(),
+        base_features=[f3],
+        idx=1,
+    )
+    fmo0.related_features = {fmo1}
+    fmo1.related_features = {fmo0}
+
+    orig_features = set([f1, f2, fadd, fmo0, fmo1])
+
+    assert len(orig_features.symmetric_difference(converted_features)) == 0
+
+
+def test_origin_feature_to_featurebase():
+    df = generate_fake_dataframe(
+        col_defs=[("idx", "Double", {"index"}), ("f_1", "Double")],
+    )
+    es = EntitySet(id="test")
+    es.add_dataframe(df, df.ww.name)
+
+    origin_features = schema_to_features(df.ww.schema)
+    f_1 = [f for f in origin_features if f.name == "f_1"][0]
+    fb = convert_feature_to_featurebase(f_1, df, {})
+
+    assert isinstance(fb, IdentityFeature)
+    assert fb.get_name() == "f_1"
+
+    f_1.name = "new name"
+    df.ww.rename({"f_1": "new name"}, inplace=True)
+    fb = convert_feature_to_featurebase(f_1, df, {})
+
+    assert isinstance(fb, IdentityFeature)
+    assert fb.get_name() == "new name"
+
+
+def test_stacked_feature_to_featurebase():
+    df = generate_fake_dataframe(
+        col_defs=[("idx", "Double", {"index"}), ("f_1", "Double")],
+    )
+    es = EntitySet(id="test")
+    es.add_dataframe(df, df.ww.name)
+
+    origin_features = schema_to_features(df.ww.schema)
+    f_1 = [f for f in origin_features if f.name == "f_1"][0]
+    features = generate_features_from_primitives([f_1], [Absolute()])
+
+    f_2 = [f for f in features if f.name == "ABSOLUTE(f_1)"][0]
+
+    fb = convert_feature_to_featurebase(f_2, df, {})
+
+    assert isinstance(fb, TransformFeature)
+    assert fb.get_name() == "ABSOLUTE(f_1)"
+    assert len(fb.base_features) == 1
+    assert fb.base_features[0].get_name() == "f_1"
+
+    f_2.name = "f_2"
+    fb = convert_feature_to_featurebase(f_2, df, {})
```

Comment:
I'm not sure if this is the right place for this question, but what happens if we create the `f_2` feature which depends on `f_1`, and then after that `f_1` gets renamed (a new alias, I think), does everything still work as expected when it comes time to calculate feature values for `f_2`? What would happen to the default name of the `f_2` feature in this case?

Reply:
Changing the alias has the biggest impact when dealing with origin features and should be avoided as much as possible, but should be fine: 

Here is what would happen:
- You create an origin Feature based on column named "Column A"
- You create an engineered Feature using the Absolute primitive to create the name "ABSOLUTE(Column A)"
- You now have 2 features, and if you want to create a feature matrix off of this, you need to convert these back to FeatureBase features first using the conversion function. This will create 2 FeatureBase features with names above and everything should work as expected

Now if you want to change the alias AFTER feature discovery, here is what would happen.
- you set an alias on the origin LiteFeature, to give it a name of "f1", and alter the corresponding column on the dataframe
- you now need to convert the 2 LiteFeatures to FeatureBase features. The Origin Feature would create an IdentityFeature pointing to the "f1" column on the dataframe. Then it would create a TransformFeature, with the name of "ABSOLUTE(Column A)", which probably isn't what you want, but it will still be pointing to the correct base_feature.
- Essentially everything works as expected, its just the names won't be right.